### DISCUSSION TITLE
Naming

### DISCUSSION THREAD

**Discussion by davebcn87** [Shopify/flash-list#111] [General]
I think we could start a discussion on whether we want or not to change the name of the component. People keeps mixing `RecyclerFlatList` and `RecyclerListView`, which can be a problem in the future.

Current name pros/cons:

## PROS
- Defines what it is, a mix of RecyclerListView and FlatList

## CONS
- Can be confused as RecyclerListView
- People have already bad concept of FlatList and containing the name could be misleading.

What are your thoughts on this?

**Comment by fortmarek** [comment]
I like the name because it clearly explains what it does. But it is easy to mistake it with `RecyclerListView`. Whatever name we choose, I think we should keep the name as `{Prefix}FlatList`. Instead of `Recycler` prefix, we could borrow terminology from `UITableView` and name our component `ReuseFlatList` (or some other derived name from the reusing concept).

If we want to change the name, we should imho act quickly because we are already spreading the name `RecyclerFlatList` and the longer we wait, the more ingrained the term will be in people's minds.

> People have already bad concept of FlatList and containing the name could be misleading.

I don't understand this argument - what do you mean by "bad concept"?

**Reply by davebcn87** [reply]
> Instead of Recycler prefix, we could borrow terminology from UITableView and name our component ReuseFlatList (or some other derived name from the reusing concept).

Is it the reusing part the most important? I like it better but I still think we should take a look on our unique selling proposition concepts: 

- Fast to set up if you come from FlatList →`Simple`, `Quick`, `Swap`, `Replacement`, `Easy`
- More performant in all platforms → `Performant`, `Fast`, `Snappy`, `Smooth`, 
- Consumes less memory and affects less to GC → `Efficient`, `Effective`, `Responsive`

Another alternative is to use a name that has nothing to do or that is not self-explanatory. 

> I don't understand this argument - what do you mean by "bad concept"?

I mean that people already associate `FlatList` with low performance, so could be misleading using that term in our library. 

> If we want to change the name, we should imho act quickly because we are already spreading the name RecyclerFlatList and the longer we wait, the more ingrained the term will be in people's minds.

Agree, even if I think the real deadline is when we open it to OSS.

**Reply by fortmarek** [reply]
> Is it the reusing part the most important?

That's a fair point - it's not, from the user perspective at least. From those three categories, `more performant in all platforms` might be the most important aspect. I was also thinking of a name such as `ZapList` as zap (⚡ ) is an emoji often used for speed (`Zap` only would work, too, but that name is already used for [this](https://github.com/uber-go/zap) repo).

**Reply by davebcn87** [reply]
now we are talking. This sounds interesting to explore. ZapList sounds good, and playing with the ⚡ emoji would help

**Reply by naqvitalha** [reply]
How about something as simple as ShopList? Also should we ask for suggestion in the react native group?

**Reply by fortmarek** [reply]
If I heard `ShopList`, I'd think it's not a general-purpose component but something specific for commerce. I'm pretty sure it would also confuse devs working on the Shop app 😄

**Reply by davebcn87** [reply]
Same here, I see `ShopList` as an app to have a list for the groceries.

**Reply by naqvitalha** [reply]
Haha, I agree. Crowdsourcing might be a good idea IMO

**Reply by fortmarek** [reply]
@naqvitalha will you post the question in the `#react-native` slack channel then?

**Comment by davebcn87** [comment]
I got a new name idea: `FlashList`. Reasoning behind the name: 
- Close to `FlatList` only need to change one `t` for a `sh` to make it work.
- Flash as a synonym of fast.
- Short name, easy to remember.
- Even if the name is close to FlatList it sounds different so it is more memorable.
- I did not find any big project with that name. Even domain names are most of them free.

**Reply by davebcn87** [reply]
@naqvitalha @fortmarek @ElviraBurchik thoughts here?

**Reply by fortmarek** [reply]
I think it's a pretty good name. I don't have a strong opinion on `FlashList` vs `ZapList`, I'd be ok with either of those.

**Reply by ElviraBurchik** [reply]
Agree, the name sounds good 👍

**Reply by naqvitalha** [reply]
It's a good name :) @davebcn87 why don't you put this on the react-native channel and take feedback and suggestions? We might get an even better name. I've very bad at names so I'm not even suggesting :P

**Comment by sbalay** [comment]
Naive idea, but have you considered `FastList`? `react-native-fast-list`

1. Mimics https://github.com/DylanVann/react-native-fast-image which is a widely used react native package
2. The name makes it quite clear which is the value proposal of the package, it's fast
3. Kinda funny how similar the words FastList / FlatList are.

**Reply by davebcn87** [reply]
@sbalay we discarded `FastList` because it is already a list component that Discord created a while ago. It is not officially open sourced, but there is a known gist where they shared the code. We prefer not adding any confusion that this is the same library.
